packages/react-component-library/src/components/RangeSlider/Slider.tsx

```diff
@@ -43,18 +50,27 @@ export const RangeSlider: React.FC<RangeSliderProps> = ({
   onUpdate,
   thresholds,
   hasPercentage,
-  displayUnit,
+  displayUnit = '',
+  formatValue,
   ...rest
 }) => {
   const [sliderValues, setSliderValues] = useState(values)
 
-  const onUpdateHandler = (newValues: ReadonlyArray<number>): void => {
-    setSliderValues(newValues)
+  const onUpdateHandler = useCallback(
```

Comment:
`useCallback` might help with performance here as `Slider` is a `PureComponent` (in our exemplar app the thresholds are updated dynamically with the values, so optimisation would be beneficial).